rearranged topbar and added more styles

ajsingh273 opened the pull request:
@calumhalcrow I added a top bar navigation using the bootstrap library. Stuart provided some feedback yesterday in the story, and I made some of the changes he requested.

One of his requests is to have the left nav (header.html) inside of the dropdown toggle. I could not get this to work. I tried adding the "collapse" classes, but it does not fit cleanly because the two stylesheets are colliding. Do you suggest an easier way of doing this? I was thinking about just adding another kind of top navbar or simply re-creating these templates using 1 bootstrap stylesheet (just so things are consistent). The dashboard layout would be a similar fit.

Once the changes are final, I can try to apply the styles in Explorer and maybe Webservices User Auth too. 

Let me know, thanks.

ajsingh273 committed:
rearranged topbar and added more styles

calumhalcrow commented:
@ajsingh273 

I'm not exactly sure what you mean by "have the left nav (header.html) inside of the dropdown toggle". Do you mean remove the left hand column completely and have the nav items be accessible via a dropdown in the top nav instead? If so, it should be achievable.

Regarding CSS conflicts: I noticed that there may be some (or even a lot of) unused CSS floating around in the project. Just going through it all and getting rid of everything that's not in use may help with such conflicts. You may also find that you don't need to add Bootstrap in order to make this work/look good. Front-end dependencies can sometimes be more trouble than they're worth.

Anyway, I'm out of time this evening, but I can spend time on this tomorrow if you'd like me to.

ajsingh273 commented:
@calumhalcrow 
I think Stuart wants to keep the APIs listed in the left column. And have only the additional resources in the top bar.  

What I am trying to do is make the items in the left column and the top bar, appear as 1 list in the mobile toggle. I will work on this some more tonight, and should have it ready for you to review before our call tomorrow. 

Yes, I will remove un-used css and files in general (I was planning to do this next). Thanks.

calumhalcrow commented:
You're storing the Github logo images with dimensions 256x256px, but rendering them in the browser with dimensions 25x25px. There are two problems with this:
- You are increasing the size of the page by serving larger than necessary images.
- You are forcing the browser to scale the image, which might not end up looking as nice as the same image scaled by software such as Photoshop.

In general, you should scale images down to the size you intend to use them at on the page.

calumhalcrow commented:
If you want the left nav to pop into the top nav when the width of that page drops below a certain threshold, this is how to do it:

You need to add the nav items to both the top nav and the left nav. When the page is wider than your threshold, you hide the nav in the top (display: none), and when the width drops below the threshold, you show the nav in the top nav, and hide the left nav.

In case you’re not already aware of how to do this: use [CSS media queries](http://www.w3schools.com/cssref/css3_pr_mediaquery.asp). Bootstrap already uses these internally, and there may be a way to hook into its workflow.

Since you’re defining the nav in two places, you should move the actual links and text out into the config file, so that you don’t have to update two places every time nav items change.

Let me know if you need help working with developerportal. If you get this working, let me look at how to add it to Explorer, as working with Ember is quite different than working with Jekyll. Anyway, it’s less important that this works with Explorer given that it’s only an internal tool.

ajsingh273 committed:
improved mobile view and removed un used files

ajsingh273 commented:
@calumhalcrow Thanks for the advice, adding the items again in the top bar was definitely the way to go. 

Here are the changes:
-added the left column items in the top bar with the correct css class 
-fixed the image sizes
-removed un-used stylesheets and js files

When adding the links in the config file, would I just list them like any other field and call it something like? {{ site.nav}}

calumhalcrow commented:
I'd glad to see you're removing unused code. Keeping things tidy helps any codebase :+1: 

For adding the nav items to the config, do something like this:

```
available_apis_nav:
  "market-research-library.html": "Market Research Library"
  "consolidated-screening-list.html": Consolidated Screening List"
   ...
```

then iterate over that in the templates (as you said, using something like `{{ site.available_apis_nav.each }}` or something). That way, you avoid putting HTML in the config file.

ajsingh273 committed:
add nav items to config file

ajsingh273 committed:
add nav items to config in different format and used in header

ajsingh273 committed:
updated site available apis

ajsingh273 committed:
fixed typo in FAQs doc

ajsingh273 commented:
@calumhalcrow 

I found a similar way to accomplish the left navigation links from the config file. Please take a look when you have a chance. 

James is going to review this for me today, so we can show Stuart these changes sooner. Also, it gives us another workday to make additional changes if needed. Thanks.

calumhalcrow commented on _config.yml:
```diff
@@ -13,3 +13,20 @@ exclude:
+- text: ITA Offices & Centers
+  url: ita-office-locations.html
+- text: Trade News & Articles
+  url: ita-trade-articles.html
```
This is totally fine @ajsingh273 :+1:
